### ISSUE TITLE
AWS S3 CTRL-C doesn't abort multipart upload when using rclone sync

### ISSUE THREAD

**Issue by jamesbraza** [rclone/rclone#4300]
#### What is the problem you are having with rclone?
I am trying to upload a 1 TB file to AWS S3 using `rclone sync`.  Since 1 TB is much larger than the default `--s3-upload-cutoff`, the upload is done via a multipart-upload.

When killing an `rclone sync` command using CTRL-C, I wondered if an abort isn't being called.  If a complete or abort isn't called on an upload, one continues to get charged by AWS for the partial upload.

I asked about this in the rclone forum, here: [AWS S3 - does killing (ctrl-c) a sync command call “abort” during multipart upload?](https://forum.rclone.org/t/aws-s3-does-killing-ctrl-c-a-sync-command-call-abort-during-multipart-upload/16839).  You can find more information there.

Turns out I was correct, abort is not called.  This is the issue the admin requested.

#### What is your rclone version (output from `rclone version`)
rclone v1.52.0

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Ubuntu 18.04.2 LTS (Bionic Beaver), 64-bit

####  Which cloud storage system are you using? (eg Google Drive)
```lang-none
type = s3
provider = aws
region = us-west-2
location_constraint = us-west-2
storage_class = DEEP_ARCHIVE
```

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`rclone sync`

**Comment by ncw** [comment]
We do this already in the onedrive backend (which could be simpler) and we should probably do it in all backends which use this pattern

- s3 multipart upload and copy
- b3 multipart upload and copy
- azureblob multipart upload and copy
- qingstor multipart upload
- others I've forgotten!

We should probably implement
- list uncompleted multipart uploads as a backend command
- `rclone cleanup` to get rid of them too

Let's see if it works first though - here cancellation on exit implemented for s3 multipart upload and multipart copy. Can you give this a go?

https://beta.rclone.org/branch/v1.52.0-023-g350ff98c-fix-4300-s3-cancel-multipart-beta/ (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I tweaked this slightly here

https://beta.rclone.org/branch/v1.52.0-024-g2a75b528-fix-4300-s3-cancel-multipart-beta/ (uploaded in 15-30 mins)

**Comment by ncw** [comment]
@jamesbraza did you try the above? Did it work?

**Comment by jamesbraza** [comment]
@ncw sorry this evaded my attention, just realized you had responded.

On my Ubuntu machine, I currently have a 1 TB file being uploaded (using `rclone`) to AWS S3 for the next week.

I will test this on my Mac.  The OS is macOS Mojave version 10.14.13 and the processor is an Intel 64-bit.  From the [download link](https://beta.rclone.org/branch/v1.52.0-024-g2a75b528-fix-4300-s3-cancel-multipart-beta/) shared above, which version should I download?

Should I use `rclone-v1.52.0-024-g2a75b528-fix-4300-s3-cancel-multipart-beta-osx-386.zip`?  I was unsure what `386` meant, and wanted to make sure.

**Comment by ncw** [comment]
This one is the best for a modern Mac

https://beta.rclone.org/branch/v1.52.0-024-g2a75b528-fix-4300-s3-cancel-multipart-beta/rclone-v1.52.0-024-g2a75b528-fix-4300-s3-cancel-multipart-beta-osx-amd64.zip

**Comment by jamesbraza** [comment]
Hello @ncw and thanks for pointing out the proper version.  Here's my output of `rclone -V`:

```
rclone v1.52.0-024-g2a75b528-fix-4300-s3-cancel-multipart-beta
- os/arch: darwin/amd64
- go version: go1.14.4
```

Using that, I verified the **use of CTRL-C now properly aborts a multipart upload!**

---

**Procedure**

1. Created a 22 GB `.zip` test file by zipping up a bunch of copies of the 1 GB file from [here](https://www.thinkbroadband.com/download)
2. Initiated the upload via:

```
rclone sync  --progress --exclude ".*" --log-file ./logs_dir/test_abort_multipart.log --log-level DEBUG ./file_dir test_aws_s3_multipart_upload_abort:jbraza-test/test-22gb
```

3. Wait for `progress` printout to indicate bytes are being transferred
    - Took about 45 seconds
4. Ran the command `aws s3api list-multipart-uploads --bucket jbraza-test` (see [AWS CLI v2 `list-multipart-uploads`](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3api/list-multipart-uploads.html)
    - Verified there was now one in-progress upload
5. Hit CTRL-C, killing the `rclone sync` command
    - One can see this reflected in the logs

```
2020/06/22 11:44:14 INFO  : Signal received: interrupt
2020/06/22 11:44:14 DEBUG : 22GB.zip: Cancelling multipart upload
2020/06/22 11:44:14 INFO  : Exiting...
```

6. Re-ran the `list-multipart-uploads` command, checking to see if the upload ID had been properly aborted. The `list-multipart-uploads` command returned that there were no in-progress uploads!

---

Aside: `386` and `amd64`

I was uncertain as to why one would use `amd64` for a Mac with an Intel processor, and this post totally cleared me up: [Difference between the i386 download and the amd64?](https://askubuntu.com/questions/54296/difference-between-the-i386-download-and-the-amd64)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.53

I've included similar fixes for

- s3
- box
- onedrive
- b2
- qingstor
